engine/runtime-with-polyglot/src/test/java/org/enso/interpreter/test/InsightForEnsoTest.java

```diff
@@ -0,0 +1,89 @@
+package org.enso.interpreter.test;
+
+import java.io.ByteArrayOutputStream;
+import java.io.OutputStream;
+import java.nio.file.Paths;
+import java.util.Map;
+import java.util.function.Function;
+import org.enso.polyglot.RuntimeOptions;
+import org.graalvm.polyglot.Context;
+import org.graalvm.polyglot.Language;
+import org.graalvm.polyglot.Source;
+import org.junit.After;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertNotNull;
+import org.junit.Before;
+import org.junit.Test;
+
+public class InsightForEnsoTest {
+    private Context ctx;
+    private AutoCloseable insightHandle;
+    private final ByteArrayOutputStream out = new ByteArrayOutputStream();
+
+    @Before
+    public void initContext() throws Exception {
+        this.ctx = Context.newBuilder()
+                .allowExperimentalOptions(true)
+                .option(
+                        RuntimeOptions.LANGUAGE_HOME_OVERRIDE,
+                        Paths.get("../../distribution/component").toFile().getAbsolutePath()
+                )
+                .logHandler(OutputStream.nullOutputStream())
+                .allowExperimentalOptions(true)
+                .allowIO(true)
+                .out(out)
+                .allowAllAccess(true)
+                .build();
+
+        var engine = ctx.getEngine();
+        Map<String, Language> langs = engine.getLanguages();
+        assertNotNull("Enso found: " + langs, langs.get("enso"));
+
+        @SuppressWarnings("unchecked")
+        var fn = (Function<Source,AutoCloseable>) engine.getInstruments().get("insight").lookup(Function.class);
```

Comment:
Btw. the following shows how to programatically register [GraalVM Insight](https://www.graalvm.org/latest/tools/graalvm-insight/) script, @hubertp.